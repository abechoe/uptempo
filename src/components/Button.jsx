import Radium from 'radium';
import React from 'react';

class Button extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <button
        type={this.props.type}
        style={styles.base}
      >
        {this.props.title}
      </button>
    );
  }
}

const styles = {
  base: {
    height: '30px',
    width: '60px',
    borderRadius: '5px',
    ':hover': {
      border: '1px solid #19d642',
      background: '#29a043',
      color: '#fff',
    },
  },
}

export default Radium(Button);