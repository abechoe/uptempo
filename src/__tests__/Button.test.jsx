import React from 'react';
import { shallow } from 'enzyme';

import Button from 'components/Button';

describe('Button component tests', () => {
  it('renders a Button component', () => {
    const wrapper = shallow(<Button type="submit" title="Submit" />);
    expect(wrapper).toMatchSnapshot();
  });
});