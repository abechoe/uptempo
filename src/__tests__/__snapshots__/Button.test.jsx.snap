// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component tests renders a Button component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    title="Submit"
    type="submit"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Submit",
      "data-radium": true,
      "onMouseEnter": [Function],
      "onMouseLeave": [Function],
      "style": Object {
        "borderRadius": "5px",
        "height": "30px",
        "width": "60px",
      },
      "type": "submit",
    },
    "ref": null,
    "rendered": "Submit",
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Submit",
        "data-radium": true,
        "onMouseEnter": [Function],
        "onMouseLeave": [Function],
        "style": Object {
          "borderRadius": "5px",
          "height": "30px",
          "width": "60px",
        },
        "type": "submit",
      },
      "ref": null,
      "rendered": "Submit",
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
